frontend/ui/uimanager.lua

```diff
@@ -377,10 +377,12 @@ function UIManager:init()
     elseif Device:isSDL() then
         self.event_handlers["Suspend"] = function()
             self:_beforeSuspend()
+            Device:onPowerEvent("Suspend")
```

Comment:
This is the onPowerEvent I referred to in an earlier comment. This function is what is needed to actually kick off the screensaver. Without this, it suspends, but no screensaver appears.

Reply:
if you change the above from `Device:isSDL()` to `Device:isEmulator()`[1] and it's probably good to go. Needs further testing though (mainly in the regular AppImage target).

[1] That's a fairly recent addition to distinguish the more recent regular desktop AppImage/Debian package from the emulator.

Reply:
So I've traced this back to the event handler for Resume not firing when I'd expect on either isSDL or isEmulator (thus not resetting the screen_saver_mode variable, and not showing a screensaver after the first attempt).

What I can do to fix this is to add another event handler in uimanager for Device:isEmulator for the default input event,  check if it's in screensaver mode, in which case do a device resume, and then send the input event along.

This seems to work:
```         self.event_handlers["__default__"] = function(input_event)
             if Device.screen_saver_mode then
                 self:resume()
            end
            self:sendEvent(input_event)
         end
```

Would you prefer I put this in a separate PR at this point? I'm not as confident in my testing of this, though it seems to work fine in my limited tests. I'd also rip out the "Suspend" and "Resume" UI messages which seem pretty useless if the screensaver triggers when expected...

Reply:
Yes, that definitely doesn't sound related and it'd just hold up this one. ;-)